pkg/operator/k8sutil/node.go

```diff
@@ -114,12 +114,6 @@ func GetNodeSchedulable(node v1.Node) bool {
 	if node.Spec.Unschedulable {
```

Comment:
From my understanding, Kubernetes sets `node.Spec.Unschedulable` if there is a `NoSchedule` taint on the node, which effectively equates to the deleted for loop below. I don't think removing this is really fixing the problem. The problem is that the check for "is the node schedulable?", likely isn't the right check for a lot of calls. Instead, those calls should probably be asking, "is the node placeable for Rook?"

`NodeMeetsPlacementTerms` exists below.

Reply:
@BlaineEXE To summarize, the current PR should fix the issue that @lbogdan encountered, right? If so we can go ahead with this PR and track a more correct solution in a separate PR. Or are there changes this PR needs to incorporate as well?